Mage/src/main/java/mage/abilities/effects/common/MeldEffect.java

```diff
@@ -46,48 +50,59 @@ public MeldEffect copy() {
-            UUID sourceId = source != null ? source.getSourceId() : null;
-            FilterControlledCreaturePermanent filter = new FilterControlledCreaturePermanent("creature named " + meldWithName);
-            filter.add(new NamePredicate(meldWithName));
-            TargetPermanent target = new TargetControlledCreaturePermanent(filter);
-            Set<UUID> meldWithList = target.possibleTargets(source.getControllerId(), source, game);
-            if (meldWithList.isEmpty()) {
-                return false; // possible permanent has left the battlefield meanwhile
-            }
-            UUID meldWithId = null;
-            if (meldWithList.size() == 1) {
-                meldWithId = meldWithList.iterator().next();
-            } else {
-                if (controller.choose(Outcome.BoostCreature, target, source, game)) {
-                    meldWithId = target.getFirstTarget();
-                }
-            }
-            // Exile the two permanents to meld.
-            Permanent sourcePermanent = game.getPermanent(sourceId);
-            Permanent meldWithPermanent = game.getPermanent(meldWithId);
-            if (sourcePermanent != null && meldWithPermanent != null) {
-                Set<Card> toExile = new HashSet<>();
-                toExile.add(sourcePermanent);
-                toExile.add(meldWithPermanent);
-                controller.moveCards(toExile, Zone.EXILED, source, game);
-                // Create the meld card and move it to the battlefield.
-                Card sourceCard = game.getExile().getCard(sourceId, game);
-                Card meldWithCard = game.getExile().getCard(meldWithId, game);
-                if (sourceCard != null && !sourceCard.isCopy() && meldWithCard != null && !meldWithCard.isCopy()) {
-                    meldCard.setOwnerId(controller.getId());
-                    meldCard.setTopHalfCard(meldWithCard, game);
-                    meldCard.setBottomHalfCard(sourceCard, game);
-                    meldCard.setMelded(true, game);
-                    game.addMeldCard(meldCard.getId(), meldCard);
-                    game.getState().addCard(meldCard);
-                    meldCard.setZone(Zone.EXILED, game);
-                    controller.moveCards(meldCard, Zone.BATTLEFIELD, source, game);
-                }
-                return true;
-            }
+        Permanent sourcePermanent = source.getSourcePermanentIfItStillExists(game);
+        if (controller == null
+                || sourcePermanent == null
+                || !sourcePermanent.isControlledBy(controller.getId())
+                || !sourcePermanent.isOwnedBy(controller.getId())) {
+            return false;
+        }
+        // Find the two permanents to meld.
+        FilterPermanent filter = new FilterControlledPermanent("permanent named " + meldWithName);
+        filter.add(new NamePredicate(meldWithName));
+        filter.add(TargetController.YOU.getOwnerPredicate());
+        Permanent meldWith;
```

Comment:
It doesn't look like this line of code is used.

Reply:
good catch, removed